src/main/java/net/spy/memcached/protocol/TCPMemcachedNodeImpl.java

```diff
@@ -483,6 +457,33 @@ public final ByteBuffer getWbuf() {
    * @see net.spy.memcached.MemcachedNode#doTlsHandshake()
    */
   public final boolean doTlsHandshake(long timeoutInMillis) throws IOException {
+    // Initialize SSLEngine and TLSConnectionHandler for TLS connection
+    SSLContext sslContext = connectionFactory.getSSLContext();
+    assert sslContext != null : "Something is wrong with TLS connection mechanism";
+
+    SSLEngine sslEngine;
+    if (!connectionFactory.skipTlsHostnameVerification() && connectionFactory.getHostnameForTlsVerification() == null){
+      throw new IllegalArgumentException("Please specify hostname for TLS verification or explicitly skip hostname verification.");
+    }
+    if (connectionFactory.getHostnameForTlsVerification() != null) {
+      // Configure SSLParameters when TLS/SSL hostname verification is enabled.
+      sslEngine = sslContext.createSSLEngine(connectionFactory.getHostnameForTlsVerification(), ((InetSocketAddress) socketAddress).getPort());
+      SSLParameters sslParams = sslEngine.getSSLParameters();
+      sslParams.setEndpointIdentificationAlgorithm("HTTPS");
+      sslEngine.setSSLParameters(sslParams);
+    } else {
+      sslEngine = sslContext.createSSLEngine();
+    }
+    sslEngine.setUseClientMode(true);
+    tlsConnectionHandler = new TLSConnectionHandler(channel, sslEngine);
+    
+    int tlsBufSize = sslEngine.getSession().getPacketBufferSize();
+    if (bufSize < tlsBufSize) {
+      // Allocate rbuf and wbuf size for TLS connections
```

Comment:
Here you also need to update `this.bufSize` to the new value, so you won't keep re-allocating the read and write buffers each time you re-connect.

Reply:
It makes sense. I will do.